src/FirmwarePlugin/FirmwarePlugin.h

```diff
@@ -150,6 +150,15 @@ class FirmwarePlugin : public QObject
     /// @return The minimum takeoff altitude (relative) for guided takeoff.
     virtual double minimumTakeoffAltitude(Vehicle* /*vehicle*/) { return 10; }
 
+    /// @return The maximum horizontal groundspeed for a multirotor.
+    virtual double maximumHorizontalSpeedMultirotor(Vehicle* /*vehicle*/) { return 10; }
+
+    /// @return The maximum equivalent airspeed setpoint.
+    virtual double maximumEquivalentAirspeed(Vehicle* /*vehicle*/) { return 25; }
+
+    /// @return The minimum equivalent airspeed setpoint
+    virtual double minimumEquivalentAirspeed(Vehicle* /*vehicle*/) { return 12; }
```

Comment:
Should these be pure-virtual?  Given that this is a fall-out-of-sky value if you get it wrong, defaults don't seem the right way to do things...

Reply:
@peterbarker If I make them pure then I will need an implementation for Ardupilot, which I think is anyway the goal.
Could you please tell me how I get these values from Ardupilot. I assume it's parameters.

I'm actually wondering if it would make more sense to have the base function return negative values which indicates that the values are not available / not valid.
Because even in the case where you get these vales from parameters, if the param fetch fails then I'd like to return invalid. In that case QGC could take action and disable the speed change button.

Reply:
> @peterbarker If I make them pure then I will need an implementation for Ardupilot, which I think is anyway the goal. 

Yep, understood.  This shouldn't be a blocker for your PR here, 'though, so maybe mark being `pure-virtual` down as "aspirational" :-)

> Could you please tell me how I get these values from Ardupilot. I assume it's parameters.

Sadly it's a bit more complicated than that.

Plane has `ARSPD_FBW_MIN` and `ARSPD_FBW_MAX`.  There's no maximum groundspeed, but there is a minimum groundspeed in `MIN_GNDSPD_CM`

Copter doesn't have a maximum speed while doing waypoint navigation; it's inferred from the (parameterised) maximum circular angle of the vehicle.  Given it's less important than the fixed-wing numbers, probably easiest to just choose a couple of constants for ArduPilot for now.  10m/s is way too slow, 'though :-)  30m/s would probably make for a reasonable upper bound guestimate.  Not suitable for some, but higher and your UI is going to suffer (almost need expo on the slider in this case!).

> I'm actually wondering if it would make more sense to have the base function return negative values which indicates that the values are not available / not valid. Because even in the case where you get these vales from parameters, if the param fetch fails then I'd like to return invalid. In that case QGC could take action and disable the speed change button.

Seems reasonable.  That also means that if the firmware behaviour changes out from underneath QGC it's going to do something sensible.  (My preference is for using NaNs in such cases; it makes it easier to catch when they get used in calculations when they shouldn't be - assuming QGC is set up for that).